src/main/java/uk/gov/hmcts/reform/security/keyvault/KeyVaultRSASignature.java

```diff
@@ -52,7 +54,21 @@ protected void engineInitSign(PrivateKey privateKey) throws InvalidKeyException
         if (!(privateKey instanceof KeyVaultRSAPrivateKey)) {
             throw new InvalidKeyException("PrivateKey must be an instance of " + KeyVaultRSAPrivateKey.class.getName());
         }
-        identifier = ((KeyVaultKey) privateKey).getIdentifier();
+        identifier = clearVersionFromKey(((KeyVaultKey) privateKey).getIdentifier());
```

Comment:
Isn't that too expensive to run on each signing operation?

Reply:
lets try this